agent/agent/config.py

```diff
@@ -0,0 +1,173 @@
+COMPILER_PORT = os.environ.get('COMPILER_PORT', '443')
+COMPILER_PROTO = os.environ.get('COMPILER_PROTO', 'https')
+COMPILER_PATH = os.environ.get('COMPILER_PATH', '')
+COMPILER_ENDPOINT = os.environ.get(
+    'COMPILER_ENDPOINT',
+    f'{COMPILER_PROTO}://{COMPILER_HOST}:{COMPILER_PORT}/{COMPILER_PATH}'
+)
+COMPILER_VERIFY = bool(int(os.environ.get('COMPILER_VERIFY', '1')))
+
+CLASSIFIER_ENDPOINT = os.environ.get('CLASSIFIER_ENDPOINT', 'http://localhost:8000')
+CLASSIFIER_VERIFY = bool(int(os.environ.get('CLASSIFIER_VERIFY', '0')))
+
+ENABLE_ASYNC = os.environ.get('ENABLE_ASYNC', '0')
+"""
+If ``1``, asynchronous callbacks will be dispatched to the worker.
+
+Otherwise they will be executed in the thread in which they are called.
+"""
+
+ENABLE_CALLBACKS = os.environ.get('ENABLE_CALLBACKS', '0')
+"""If ``0``, callbacks bound to events will not be executed."""
+
+JWT_SECRET = os.environ.get('JWT_SECRET')
+
+CORE_VERSION = "0.0.0"
+
+# Email notification configuration.
+EMAIL_ENABLED = bool(int(os.environ.get('EMAIL_ENABLED', '1')))
+DEFAULT_SENDER = os.environ.get('DEFAULT_SENDER', 'noreply@arxiv.org')
+SUPPORT_EMAIL = "help@arxiv.org"
+SMTP_HOSTNAME = os.environ.get('SMTP_HOSTNAME', 'localhost')
+SMTP_USERNAME = os.environ.get('SMTP_USERNAME', 'foouser')
+SMTP_PASSWORD = os.environ.get('SMTP_PASSWORD', 'foopass')
+SMTP_PORT = int(os.environ.get('SMTP_PORT', '0'))
+SMTP_LOCAL_HOSTNAME = os.environ.get('SMTP_LOCAL_HOSTNAME', None)
+SMTP_SSL = bool(int(os.environ.get('SMTP_SSL', '0')))
+
+
+EXTERNAL_URL_SCHEME = os.environ.get('EXTERNAL_URL_SCHEME', 'https')
+BASE_SERVER = os.environ.get('BASE_SERVER', 'arxiv.org')
+SERVER_NAME = "submit.arxiv.org"
+
+URLS = [
+    ("submission", "/<int:submission_id>", "submit.arxiv.org"),
+    ("confirmation", "/<int:submission_id>/confirmation", "submit.arxiv.org")
+]
+"""
+URLs for external services, for use with :func:`flask.url_for`.
+This subset of URLs is common only within submit, for now - maybe move to base
+if these pages seem relevant to other services.
+
+For details, see :mod:`arxiv.base.urls`.
+"""
+
+
+MAX_SAVE_RETRIES = 25
+DEFAULT_SAVE_RETRY_DELAY = 30
+
+
+# TODO: make this configurable
+RECLASSIFY_PROPOSAL_THRESHOLD = 0.57   # Equiv. to logodds of 0.3.
+"""This is the threshold for generating a proposal from a classifier result."""
+
+# TODO: make this configurable.
+LOW_STOP_PERCENT = 0.10
+"""This is the threshold for abornmally low stopword content by percentage."""
+
+LOW_STOP = 400
+"""This is the threshold for abornmally low stopword content by count."""
+
+HIGH_STOP_PERCENT = 0.30
+"""This is the threshold for abnormally high stopword content by percentage."""
+
+MULTIPLE_LIMIT = 1.01
+LINENOS_LIMIT = 0
+
+TITLE_SIMILARITY_WINDOW = 3*365/12    # days
+TITLE_SIMILARITY_THRESHOLD = 0.7
+METADATA_ASCII_THRESHOLD = 0.5
+
+
+COMPRESSED_PACKAGE_MAX = 6_000_000
+UNCOMPRESSED_PACKAGE_MAX = 18_000_000
+PDF_LIMIT = 15_000_000
+"""The maximum size of the resulting PDF."""
```

Comment:
```suggestion
"""The maximum size in bytes of the resulting PDF."""
```
I would also consider renaming the above variables to include the unit.

Reply:
Renamed variables in 57cca44